### PR TITLE
Type cleanups

### PR DESCRIPTION
**Pull request by mr-c**
Also added makefile targets: pydocstyle_report diff_pydocstyle_report

Some material from https://github.com/DataBiosphere/toil/pull/3493

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Many type cleanups, cwltoil.py is now checked strictly
 
## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -121,8 +121,8 @@ clean_sdist:
 # We always claim to be Travis, so that local test runs will not skip Travis tests.
 # Setting SET_OWNER_TAG will tag cloud resources so that UCSC's cloud murder bot won't kill them.
 test: check_venv check_build_reqs
-	TRAVIS=true TOIL_OWNER_TAG="shared" \
-	    python -m pytest --durations=0 --log-level DEBUG --log-cli-level INFO -r s $(cov) $(tests)
+	TRAVIS=true \
+	    python -m pytest --cov=toil --durations=0 --log-level DEBUG --log-cli-level INFO -r s $(tests)
```
Comment (mr-c) [inline Makefile:125]: Whoops, I'll remove this on the next commit

File: Makefile [hunk 2]
```diff
@@ -121,8 +121,8 @@ clean_sdist:
 # We always claim to be Travis, so that local test runs will not skip Travis tests.
 # Setting SET_OWNER_TAG will tag cloud resources so that UCSC's cloud murder bot won't kill them.
 test: check_venv check_build_reqs
-	TRAVIS=true TOIL_OWNER_TAG="shared" \
-	    python -m pytest --durations=0 --log-level DEBUG --log-cli-level INFO -r s $(cov) $(tests)
+	TRAVIS=true \
```
Comment (DailyDreaming) [inline Makefile:124]: We need `TOIL_OWNER_TAG="shared"` so that our AWS bot doesn't kill any AWS resources we create.

**Review by DailyDreaming — CHANGES_REQUESTED** [review]
@mr-c Thanks for doing this!  See comment below.

File: src/toil/test/docs/scriptsTest.py [hunk 3]
```diff
@@ -162,10 +160,6 @@ def testPromises(self):
     def testServices(self):
         self.checkExitCode("tutorial_services.py")
 
-    @travis_test
-    def testStaging(self):
-        self.checkExitCode("tutorial_staging.py")
-
```
Comment (mr-c) [inline src/toil/test/docs/scriptsTest.py:168]: This was accidental, will restore

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Thank you @mr-c .

**Pull request merged**